### ISSUE TITLE
WAUT GSP switching points

### ISSUE THREAD

**Issue by malerczyk** [eclipse/sumo#7392]
Hi,

I want to simulate a WAUT.
I know, that the GSP parameter shall determine the time for switching my programs, given any logic.
However this does not work for me.
Instead of the GSP-switches I'm getting immediate hard switches as I marked in the Tracker-Screenshot.
![WAUT_SUMO](https://user-images.githubusercontent.com/69584566/90013513-b1449d80-dca5-11ea-968c-455585028634.png)
I also don't unterstand, how the switching times come about.
My cycle is 90s, so the first switch at about 360s makes some sense, because it is four times my cycle time.
But still this is by no means a GSP-Point.
Is there an option to manually specify the GSP point?

My tlLogic looks like this:
```
<tlLogic id="281" type="static" programID="17" offset="0">
        <phase duration="5"  state="OGGrrrrGG"/>
        <phase duration="1"  state="OyGrrrrGG"/>
        <phase duration="2"  state="OyGrrrrGr"/>
        <phase duration="4"  state="OrGrrrrGr"/>
        <phase duration="1"  state="OrGurrrGr"/>
        <phase duration="14" state="OrGGrrrGr"/>
        <phase duration="3"  state="OryGrrrGr"/>
        <phase duration="8"  state="OrrGrrrGr"/>
        <phase duration="2"  state="Orryrrrrr"/>
        <phase duration="1"  state="OrryrrGrr"/>
        <phase duration="3"  state="OrrrrrGrr"/>
        <phase duration="1"  state="OrrruuGrr"/>
        <phase duration="3"  state="OrrrGgGrr"/>
        <phase duration="12" state="OrrrGGrrr"/>
        <phase duration="5"  state="GrrrGGrrr"/>
        <phase duration="3"  state="Grrryyrrr"/>
        <phase duration="1"  state="yrrrrrrrr"/>
        <phase duration="1"  state="yrurrrrrr"/>
        <phase duration="1"  state="yrGrrrrGr"/>
        <phase duration="2"  state="OrGrrrrGr"/>
        <phase duration="1"  state="OuGrrrrGr"/>
        <phase duration="16" state="OGGrrrrGG"/>
    </tlLogic>
    <tlLogic id="281" type="static" programID="18" offset="0">
        <phase duration="1"  state="yrrrrrrrr"/>
        <phase duration="1"  state="yrurrrrrr"/>
        <phase duration="1"  state="yrGrrrrGr"/>
        <phase duration="2"  state="OrGrrrrGr"/>
        <phase duration="1"  state="OuGrrrrGr"/>
        <phase duration="21" state="OGGrrrrGG"/>
        <phase duration="1"  state="OyGrrrrGG"/>
        <phase duration="2"  state="OyGrrrrGr"/>
        <phase duration="4"  state="OrGrrrrGr"/>
        <phase duration="1"  state="OrGurrrGr"/>
        <phase duration="14" state="OrGGrrrGr"/>
        <phase duration="3"  state="OryGrrrGr"/>
        <phase duration="8"  state="OrrGrrrGr"/>
        <phase duration="2"  state="Orryrrrrr"/>
        <phase duration="1"  state="OrryrrGrr"/>
        <phase duration="3"  state="OrrrrrGrr"/>
        <phase duration="1"  state="OrrruuGrr"/>
        <phase duration="3"  state="OrrrGgGrr"/>
        <phase duration="12" state="OrrrGgrrr"/>
        <phase duration="5"  state="GrrrGgrrr"/>
        <phase duration="3"  state="Grrryyrrr"/>
 </tlLogic>
```
and my WAUT like this:
```
    <WAUT refTime="0" id="LSA_281" startProg="18">
         <wautSwitch time="300" to="17"/>
         <wautSwitch time="600" to="18"/>
    </WAUT>
    <wautJunction wautID="LSA_281" junctionID="281" procedure="GSP"/>
```

I also tried the "synchron" parameter in the wautJunction command but this didn't change anything.
Could someone please explain what exactly this parameter does?

Thanks!

**Comment by behrisch** [comment]
You can change the GSP by giving it as a parameter to the logic.
```
<tlLogic>
    ...
    <parameter key="GSP" value="40"/>
</tlLogic>
I think synchron only has an effect if you change multiple junctions with the same WAUT, but I am not 100% sure.

**Comment by malerczyk** [comment]
Hey, 
I tried your suggestion, but it does not work.
Setting the parameter does not have any effect at all.

By reading the source code I found that 
``MSTLLogicControl::WAUTSwitchProcedure::getGSPValue(const MSTrafficLightLogic& logic)``
always returns 0. 
This is caused by ``std::string val = logic.getParameter("GSP", "");`` which does not seem to find the "GSP" key and therefore always returns an empty string.
However, I was not able to find the cause of this.
